boards/wio_terminal/src/sensors.rs

```diff
@@ -9,24 +10,31 @@ use atsamd_hal::sercom::{I2CMaster4, PadPin, Sercom4Pad0, Sercom4Pad1};
 use lis3dh::{Lis3dh, SlaveAddr};
 
 /// I2C Accelerometer pins (uses `SERCOM4`)
-pub struct Accelerometer {
+pub struct Accelerometer<C, D>
+where
+    C: Into<I2c0Scl>,
+    D: Into<I2c0Sda>,
+{
     /// `I2C0` bus clock pin
-    pub scl: Pa12<Input<Floating>>,
+    pub scl: C,
 
     /// `I2C0` bus data pin
-    pub sda: Pa13<Input<Floating>>,
+    pub sda: D,
 }
```

Comment:
Shouldn't the `struct` hold `I2c0Scl` and `I2cSda` directly, instead of being generic? You only need `Accelerometer::init` to be generic

Reply:
Oh wait, this is the whole "Sets" approach. In that case, it still shouldn't be generic, but instead should hold `Pin<PA12, Disabled<Floating>>`, otherwise known as `Pin<PA12, Reset>`. Or you could do like I did in the `pygamer` PR and add an additional alias as `I2cSclReset` or something.